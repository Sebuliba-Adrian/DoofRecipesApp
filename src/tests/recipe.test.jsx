import React from "react";
import renderer from "react-test-renderer";
import Recipe from "../components/recipes/recipe";

const props = {
  key: 1,
  selectedCategory: {
    id: 1,
    name: "Selected Category",
    description: "This is the selected category",
    date_created: "Sun, 11 Feb 2018 11:31:50 GMT",
    date_modified: "Sun, 11 Feb 2018 11:31:50 GMT"
  },
  recipe: {
    id: 1,
    name: "Test Recipe",
    description: "This is a test recipe",
    date_created: "Sun, 11 Feb 2018 11:31:50 GMT",
    date_modified: "Sun, 11 Feb 2018 11:31:50 GMT",
   
  },
  request: jest.fn()
};
it("renders correctly", () => {
  const recipe = renderer.create(<Recipe {...props} />).toJSON();
  expect(recipe).toMatchSnapshot();
});
