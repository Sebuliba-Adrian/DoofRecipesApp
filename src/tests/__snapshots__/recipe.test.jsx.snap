// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-6"
>
  <div
    className="card animated zoomIn"
  >
    <div
      className="card-block"
    >
      <div
        className="card-header"
      >
        <ul
          className="nav nav-pills card-header-pills text-center"
        >
          <div
            aria-label="Basic example"
            className="btn-group float-right"
            role="group"
          >
            <a
              className="card-link"
              data-backdrop="static"
              data-keyboard="false"
              data-target="#1"
              data-toggle="modal"
              href="#/"
            >
              Edit
            </a>
            <a
              className="card-link"
              data-backdrop="static"
              data-keyboard="false"
              data-target="#deleteRecipeModel1"
              data-toggle="modal"
              href="#/"
            >
              Delete
            </a>
          </div>
        </ul>
      </div>
      <h5
        className="card-title"
      >
        Test Recipe
      </h5>
      <p
        className="card-text"
        style={
          Object {
            "whiteSpace": "pre-wrap",
          }
        }
      >
        This is a test recipe
      </p>
    </div>
  </div>
  <div
    className="modal fade"
    id={1}
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="exampleModalLabel"
          />
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
          <div
            className="input-group"
            onChange={[Function]}
          >
            <input
              className="form-control"
              id="name"
              name="name"
              onChange={[Function]}
              type="text"
              value="Test Recipe"
            />
          </div>
          <br />
          <label
            htmlFor="description"
          >
            Description
          </label>
          <div
            className="input-group"
          >
            <textarea
              className="form-control"
              id="description"
              maxLength="140"
              name="description"
              onChange={[Function]}
              rows="7"
              type="text"
              value="This is a test recipe"
            />
          </div>
          <br />
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-primary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id="deleteRecipeModel1"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="exampleModalLabel"
          >
            Delete
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          Are you sure you want to delete?
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            id="cancel"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-primary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
